### PR TITLE
Second attempt at adding skeletal model settings to protocol

### PR DESCRIPTION
**Pull request by TheButlah**
A revised version of #30

I had the following goals which prompted a new PR:
* Make it more clear what the datatypes of settings were
* Don't introduce a whole table just to associate a setting with a value for each datatype
* Clarify the naming
* Switch from calling things "FkSettings" to "ModelSettings". This is because in the future, we will have settings that aren't entirely just forward kinematics, since we may also support positional constraints from vive trackers

### CODE REVIEW

**Comment by TheButlah** [comment]
Please don't merge yet, consider this a request for comments

**Comment by TheButlah** [comment]
One other idea worth mentioning:
We are just representing fields and their settings. Why even bother with enums at that point? Isn't it better to just have a single table, with fields.

**Comment by TheButlah** [comment]
> One other idea worth mentioning: We are just representing fields and their settings. Why even bother with enums and associating those enums to values? Isn't it better to just have a single table, with fields.

Discussed with @loucass003 and we redid it a third time, lol

File: protocol/flatbuffers/settings/model.fbs
Comment (TheButlah) [inline protocol/flatbuffers/settings/model.fbs:18]: We should name this clearly, before merging

Comment (Louka3000) [reply protocol/flatbuffers/settings/model.fbs:18]: `waist_spine_averaging` is the waistNode interpolating between the chest and hip when the waist tracker is missing.
`waist_pelvis_averaging` is the waist interpolating between the chest and pelvis* when the waist tracker is missing.
`hip_pelvis_averaging` is the hip interpolating between a spine tracker (waist or chest) and the pelvis* when the hip tracker is missing.

`pelvis_hip_averaging` is the hipNode (not tracker) interpolating between itself and the pelvis*
`waist_tracker_pelvis_averaging` is the trackerWaistNode interpolating between itself and the pelvis on its yaw and roll only.

`knee_tracker_angle_averaging` is the kneeTrackerNodes interpolating between themselves and the ankles on their yaw and roll only.

*Pelvis = average of the two upper leg trackers.

File: protocol/flatbuffers/settings/model.fbs
Comment (TheButlah) [inline protocol/flatbuffers/settings/model.fbs:5]: reminder to self: check to ensure these are optional, so that I don't need to specify all settings when I modify settings

**Review by loucass003 — APPROVED** [review]

**Review by Louka3000 — COMMENTED** [review]
~~I renamed ModelRatios to be clearer.
naming rule: toBeChanged_changingTowards_type.
For example, waist_hip_averaging means that if it's 0, waist will == the tracker right abow it (chest). If it's 1, waist will == hip.~~

File: protocol/flatbuffers/settings/model.fbs
Comment (Louka3000) [reply protocol/flatbuffers/settings/model.fbs:5]: Oh ya, probably important to do that....

Comment (Louka3000) [reply protocol/flatbuffers/settings/model.fbs:5]: Same with the ratios btw!

**Review by TheButlah — COMMENTED** [review]
Reviewed and fixed the names with @Louka3000 and they should be good now. Also made those fields optional so not all settings must be given at first.

Please hold off on merging until @Louka3000 has a chance to test, because we omitted the `pelvid_averaging` setting since it didn't seem to be needed.

**Comment by Louka3000** [comment]
> Please hold off on merging until @Louka3000 has a chance to test, because we omitted the `pelvis_averaging` setting since it didn't seem to be needed.

I wasn't happy with what we had by removing the pelvis averaging, as, if we wanted the pelvis to follow the legs more, they also followed them on pitch.
So, I'm renaming the fifth value from WaistTrackerLegsAveraging to HipLegsAveraging. The old value rotated the waist tracker on its yaw and roll. Now the new value does the same for the tracker and hipNode.

Basically merging the pelvisLegs averaging the the waistTrackerPelvis averaging into a single value.

**Review by Louka3000 — APPROVED** [review]
These toggles and ratios satisfy me.
Idk how to go about solving the conflict tho, as they're generated files...

**Review by Louka3000 — APPROVED** [review]
uwu

**Review by TheButlah — COMMENTED** [review]
lgtm, but I made the pr so I can't approve

**Pull request merged**
